paymentsheet-example/src/main/java/com/stripe/android/paymentsheet/example/samples/ui/customersheet/playground/CustomerSheetPlaygroundViewModel.kt

```diff
@@ -0,0 +1,329 @@
+package com.stripe.android.paymentsheet.example.samples.ui.customersheet.playground
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import com.github.kittinunf.fuel.Fuel
+import com.github.kittinunf.fuel.core.FuelError
+import com.github.kittinunf.fuel.core.extensions.jsonBody
+import com.github.kittinunf.fuel.core.requests.suspendable
+import com.github.kittinunf.result.Result
+import com.stripe.android.PaymentConfiguration
+import com.stripe.android.customersheet.CustomerAdapter
+import com.stripe.android.customersheet.CustomerEphemeralKey
+import com.stripe.android.customersheet.CustomerSheet
+import com.stripe.android.customersheet.CustomerSheetResult
+import com.stripe.android.customersheet.ExperimentalCustomerSheetApi
+import com.stripe.android.paymentsheet.PaymentSheet
+import com.stripe.android.paymentsheet.example.samples.networking.ExampleCreateSetupIntentRequest
+import com.stripe.android.paymentsheet.example.samples.networking.ExampleCreateSetupIntentResponse
+import com.stripe.android.paymentsheet.example.samples.networking.ExampleCustomerSheetRequest
+import com.stripe.android.paymentsheet.example.samples.networking.ExampleCustomerSheetResponse
+import com.stripe.android.paymentsheet.example.samples.networking.awaitModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import kotlinx.serialization.json.Json
+
+@OptIn(ExperimentalCustomerSheetApi::class)
+class CustomerSheetPlaygroundViewModel(
+    application: Application,
+) : AndroidViewModel(application) {
+
+    private val _viewState =
+        MutableStateFlow<CustomerSheetPlaygroundViewState>(CustomerSheetPlaygroundViewState.Loading)
+    val viewState: StateFlow<CustomerSheetPlaygroundViewState> = _viewState
+
+    private val initialConfiguration = CustomerSheet.Configuration.Builder()
+        .defaultBillingDetails(
+            PaymentSheet.BillingDetails(
+                name = "CustomerSheet Testing"
+            )
+        ).billingDetailsCollectionConfiguration(
+            PaymentSheet.BillingDetailsCollectionConfiguration(
+                name = PaymentSheet.BillingDetailsCollectionConfiguration.CollectionMode.Always
+            )
+        )
+        .googlePayEnabled(viewState.value.isGooglePayEnabled)
+        .build()
+    val configuration: StateFlow<CustomerSheet.Configuration> = viewState.map {
+        CustomerSheet.Configuration.Builder()
```

Comment:
Nit: Did we discuss adding a `configuration.newBuilder(): Configuration.Builder` method?

With that, we could just do the following here:

```kotlin
val configuration: StateFlow<CustomerSheet.Configuration> = viewState.map {
    initialConfiguration
        .newBuilder()
        .googlePayEnabled(it.isGooglePayEnabled)
        .build()
}
```

Reply:
We haven't, but that is a good idea.